utils/firestore.js

```diff
@@ -2148,26 +2148,43 @@ const checkCollectionUniqueness = async (supplyId, collectionId) => {
+    `${fieldMapping.collectionDetails.toString()}.${fieldMapping.baseline.toString()}`,
+    fieldMapping.firstName.toString(),
+    fieldMapping.lastName.toString(),
+    fieldMapping.address1.toString(),
+    fieldMapping.address2.toString(),
+    fieldMapping.city.toString(),
+    fieldMapping.state.toString(),
+    fieldMapping.zip.toString(),
+    'Connect_ID',
+];
+
 /**
  * Creates new array based on query below.
  * Each participant object is transformed into a new object or an empty array.
  * @returns {Array} - Array of object(s) and/or array(s) based on processParticipantData function.
  * Ex. [{first_name: 'John', last_name: 'Doe', address_1: '123 Main St', address_2: '', city: 'Anytown', state: 'NY', zip_code: '12345', connect_id: 123457890}, ...]
  */
-const queryTotalAddressesToPrint = async () => {
+// TODO: Suboptimal process. Would benefit from direct query on a {kitStatus: initialized} or {completed: no} variable, which could be assigned on kit creation in Biospecimen.
+// This will get progressively slower and more expensive as participants are added.
+// TODO: A sliding time window would be more efficient in the .where(<timestamp>) query.
+const queryHomeCollectionAddressesToPrint = async () => {
     try {
         const { withdrawConsent, participantDeceasedNORC, activityParticipantRefusal, baselineMouthwashSample, 
             collectionDetails, baseline, bloodOrUrineCollected, bloodOrUrineCollectedTimestamp, yes, no } = fieldMapping;
 
         const snapshot = await db.collection('participants')
-        .where(withdrawConsent.toString(), '==', no)
-        .where(participantDeceasedNORC.toString(), '==', no)
-        .where(`${activityParticipantRefusal}.${baselineMouthwashSample}`, '==', no)
-        .where(`${collectionDetails}.${baseline}.${bloodOrUrineCollected}`, '==', yes)
-        .where(`${collectionDetails}.${baseline}.${bloodOrUrineCollectedTimestamp}`, '>=', '2024-04-01T00:00:00.000Z')
-        .orderBy(`${collectionDetails}.${baseline}.${bloodOrUrineCollectedTimestamp}`, 'desc')
-        .get();
-        return snapshot.docs.map(document => processParticipantData(document.data(), true));
+            .select(...participantHomeCollectionKitFields)
+            .where(withdrawConsent.toString(), '==', no)
+            .where(participantDeceasedNORC.toString(), '==', no)
+            .where(`${activityParticipantRefusal}.${baselineMouthwashSample}`, '==', no)
+            .where(`${collectionDetails}.${baseline}.${bloodOrUrineCollected}`, '==', yes)
+            .where(`${collectionDetails}.${baseline}.${bloodOrUrineCollectedTimestamp}`, '>=', '2024-04-01T00:00:00.000Z')
+            .orderBy(`${collectionDetails}.${baseline}.${bloodOrUrineCollectedTimestamp}`)
+            .get();
+
+        return snapshot.docs.map(document => processParticipantHomeMouthwashKitData(document.data(), true));
     } catch (error) {
         console.error(error);
         return new Error(error);
```

Comment:
This returned Error instance is not handled by calling function.

Reply:
same as above...updated/handled.